src/functions.c

```diff
@@ -370,6 +370,34 @@ void fcallCommandReadOnly(client *c) {
     fcallCommandGeneric(c, 1);
 }
 
+void functionsFlushCommand(client *c) {
+    if (c->argc > 3) {
+        addReplyErrorFormat(c,"wrong number of arguments for '%s' command or subcommand", c->cmd->name);
+        return;
+    }
+    int async = 0;
+    if (c->argc == 3 && !strcasecmp(c->argv[2]->ptr,"sync")) {
+        async = 0;
+    } else if (c->argc == 3 && !strcasecmp(c->argv[2]->ptr,"async")) {
+        async = 1;
+    } else if (c->argc == 2) {
+        async = server.lazyfree_lazy_user_flush ? 1 : 0;
+    } else {
+        addReplyError(c,"FUNCTION FLUSH only support SYNC|ASYNC option");
+        return;
+    }
+    functionsCtx* f_ctx = functionsCtxCreate();
+    if (async) {
+        functionsCtx* old_f_ctx = functions_ctx;
+        functions_ctx = f_ctx;
+        freeFunctionsAsync(old_f_ctx);
+    } else {
+        functionsCtxSwapWithCurrent(f_ctx);
```

Comment:
why not use `functionsCtxClear` in this branch instead create + swap?
seems less confusing..

Reply:
Right, fixed.